lib/chef/provider/dsc_resource.rb

```diff
@@ -92,6 +101,10 @@ def supports_refresh_mode_enabled?
         Chef::Platform.supports_refresh_mode_enabled?(node)
       end
 
+      def module_usage_valid?
+        !(!@module_name && @module_version) ? true : false
```

Comment:
@smurawski I will modify this.

@btm The scenarios I have handled here is like below:

| Module Name | Module Version | Result |
| ------------------ | --------------------- | -------- |
| Given              | Given                 | Valid   |
| Not given        | Not given           | Valid    |
| Given              | Not given           | Valid   |
| Not given        | Given                  | Invalid |

1. If **module_name** and **module_version** both are given then it invokes the given module with the given version.
2. If both are not given then default module **PSDesiredStateConfiguration** is used.
3. If **module_name** is given and **module_version** is not given, then it invokes the given module without the version. In this case, if more than one version of the given module is installed then Chef raises exception saying resource not found.
4. If **module_name** is not given and **module_version** is given, then this is an invalid case.

Let me know if you want to modify these scenarios. You can also take a look at the test scenarios document (attached on the card-434) to understand all the positive and negative scenarios with their results.